docs/fundamentals/code-analysis/quality-rules/ca1506.md

```diff
@@ -69,6 +69,28 @@ dotnet_analyzer_diagnostic.category-Maintainability.severity = none
 
 For more information, see [How to suppress code analysis warnings](../suppress-warnings.md).
 
+## Configure threshold
+
+You can configure the threshold at which this rule fires.
+
+1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+
+2. Add the desired threshold to the text file in the following format:
+
+   ```txt
+   CA1506: 120
+   ```
+
+   In this example, the rule is configured to fire when a type has more than 120 unique type references.
```

Comment:
I'm not sure this is correct, since this rule has different thresholds for types and methods. Is there a nuanced way to configure this rule?

Reply:
We allow defining a global threshold for all symbol kinds, or specific to a particular symbol kind. https://github.com/dotnet/roslyn-analyzers/blob/63367165996d9dedbecf6677e3315a470a2f557c/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzerTests.cs#L994-L1004

https://github.com/dotnet/roslyn-analyzers/blob/63367165996d9dedbecf6677e3315a470a2f557c/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs#L323-L325

https://github.com/dotnet/roslyn-analyzers/blob/63367165996d9dedbecf6677e3315a470a2f557c/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs#L239-L257

Reply:
Allowed SymbolKind strings are defined here: https://github.com/dotnet/roslyn-analyzers/blob/63367165996d9dedbecf6677e3315a470a2f557c/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs#L384-L406

Not all symbol kinds may be applicable for all code metrics rules though.